test/spoolchanges.js

```diff
@@ -12,179 +12,182 @@
-            };
-          })
-          : sparseResultsArray,
-        pending: pending,
+      return JSON.stringify({
+        results: fullResponse ? getResults(batchSize, seq) : sparseResultsArray,
+        pending,
         last_seq: `${lastSeq}-abc`
-      };
-    });
-}
-
-describe('#unit Check spool changes', function() {
-  it('should terminate on request error', async function() {
-    nock(url)
-      .post(`/${dbName}/_changes`)
-      .query(true)
-      .times(3)
-      .replyWithError({ code: 'ECONNRESET', message: 'socket hang up' });
-
-    return new Promise((resolve, reject) => {
-      changes(db, '/dev/null', 500, null, function(err) {
-        try {
-          assert.strictEqual(err.name, 'SpoolChangesError');
-          assert.strictEqual(err.message, `Failed changes request - socket hang up: post ${url}/${dbName}/_changes`);
-          assert.ok(nock.isDone());
-          resolve();
-        } catch (err) {
-          reject(err);
-        }
       });
-    });
-  }).timeout(longTestTimeout);
-
-  it('should terminate on bad HTTP status code response', async function() {
-    nock(url)
-      .post(`/${dbName}/_changes`)
-      .query(true)
-      .times(3)
-      .reply(500, function(uri, requestBody) {
-        this.req.response.statusMessage = 'Internal Server Error';
-        return { error: 'foo', reason: 'bar' };
-      });
-    return new Promise((resolve, reject) => {
-      changes(db, '/dev/null', 500, null, function(err) {
-        try {
-          assert.strictEqual(err.name, 'HTTPFatalError');
-          assert.strictEqual(err.message, `500 Internal Server Error: post ${url}/${dbName}/_changes - Error: foo, Reason: bar`);
-          assert.ok(nock.isDone());
-          resolve();
-        } catch (err) {
-          reject(err);
-        }
-      });
-    });
-  }).timeout(longTestTimeout);
-
-  it('should keep collecting changes', async function() {
-    // This test validates that spooling will correctly
-    // continue across multiple requests
-    // (4 batches of 100000 to be precise).
-    // This test might take up to 10 seconds
-    this.timeout(10 * 1000);
-
-    // Use full changes for this test
-    provideChanges(100000, 400000, true);
-    return new Promise((resolve, reject) => {
-      changes(db, '/dev/null', 500, null, function(err) {
-        try {
-          assert.ok(!err);
-          assert.ok(nock.isDone());
-          resolve();
-        } catch (err) {
-          reject(err);
-        }
+    }
+
+    describe('#unit shorter spool changes checks', function() {
+      it('should keep collecting changes', async function() {
+      // This test validates that spooling will correctly
+      // continue across multiple requests
+      // (4 batches of 100000 to be precise).
+      // This test might take up to 10 seconds
+        this.timeout(10 * 1000);
+
+        // Use full changes for this test
+        batchSize = 10000;
+        totalChanges = 40000;
```

Comment:
The old `batchSize` was `100000` and `totalChanges` was `400000`.  Is the change because we already have a longer test below that uses a `batchSize` of `100000`?

Reply:
accidental, fixed in https://github.com/IBM/couchbackup/pull/650/commits/c2437453687d32655c8cfd0e236efcfc7d56803a along with the comment around errorTolerance. I also opened https://github.com/IBM/cloudant-node-sdk/issues/1384 for the weird behaviour with `null`.